emcc.py

```diff
@@ -3493,18 +3493,33 @@ def un_src(self):
     """Use this if you want to modify the script and need it to be inline."""
     if self.src is None:
       return
-    self.inline = '''
-          var script = document.createElement('script');
-          script.src = "%s";
-          document.body.appendChild(script);
-''' % self.src
+    if settings.EXPORT_ES6:
+      template = '''
+        import("./%s").then(exports => exports.default(Module))
+      '''
+    else:
+      template = '''
+            var script = document.createElement('script');
+            script.src = "%s";
+            document.body.appendChild(script);
+      '''
+    self.inline = template % self.src
     self.src = None
 
   def replacement(self):
     """Returns the script tag to replace the {{{ SCRIPT }}} tag in the target"""
     assert (self.src or self.inline) and not (self.src and self.inline)
     if self.src:
-      return '<script async type="text/javascript" src="%s"></script>' % quote(self.src)
+      if settings.EXPORT_ES6:
+        template = '''
+        <script type="module">
+          import initModule from "./%s";
+          initModule(Module);
```

Comment:
I don't really know what `script type="module"` is doing..  I guess I should read up it.

Reply:
It tells browser that script is actually an ESM, and should execute in that mode (strict mode + allowing `import` declarations).